cmd_watch_dir.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// CmdWatchDir is `direnv watch-dir SHELL PATH`
+var CmdWatchDir = &Cmd{
+	Name:    "watch-dir",
+	Desc:    "Recursively adds a directory to the list that direnv watches for changes",
+	Args:    []string{"[SHELL]", "[DIR]"},
```

Comment:
```suggestion
	Args:    []string{"SHELL", "DIR"},
```
Let's make both arguments required to simplify the logic. Potentially we want to add a variable number of directories in the future to watch multiple folders at once.